File: app/src/main/java/com/suzei/minote/data/NotesDao.java
package com.suzei.minote.data;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Query;

@Dao
public interface NotesDao {

    @Query("SELECT * FROM NOTES")
    List<Notes> findAllNotes();

    @Query("SELECT * FROM NOTES WHERE _id=:itemId")
    Notes findNoteById(int itemId);

    @Delete
    void deleteNote(Notes note);

}

File: app/src/main/java/com/suzei/minote/data/local/dao/NotesDao.kt
package com.suzei.minote.data.local.dao

import androidx.room.*
import com.suzei.minote.data.local.entity.Notes

@Dao
interface NotesDao {

    @Query("SELECT * FROM NOTES")
    fun findAllNotes(): List<Notes>

    @Query("SELECT * FROM NOTES WHERE _id=:itemId")
    fun findNoteById(itemId: String): Notes

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveNote(note: Notes)

    @Update
    fun updateNote(note: Notes)

    @Delete
    fun deleteNote(note: Notes)

}
